--- grub2.spec.orig	2013-08-22 09:40:35.756617366 +0000
+++ grub2.spec	2013-08-22 09:39:46.239995789 +0000
@@ -554,6 +554,7 @@
 BuildRequires:	texinfo
 BuildRequires:	dejavu-sans-fonts
 BuildRequires:	help2man
+BuildRequires:  zfs-devel
 %ifarch %{efiarchs}
 BuildRequires:	pesign >= 0.99-8
 %endif
@@ -655,6 +656,7 @@
         --with-platform=efi					\
 	--with-grubdir=%{name}					\
         --program-transform-name=s,grub,%{name},		\
+	--enable-libzfs						\
 	--disable-grub-mount					\
 	--disable-werror
 make %{?_smp_mflags}
